IntroToCSharp/Pages/Lessons/Collections/Arrays.razor

```diff
@@ -0,0 +1,379 @@
+@inherits CaptainCoder.Page
+@page "/lessons/collections/arrays"
+
+<Title>Arrays</Title>
+
+
+<Section Title="Introdution">
+    <MudText>
+        An array is a data structure that stores a pre-defined number of variables of the same data type within its bounds.
+         The important part of this definition to take note of is that an array is predefined, meaning that you have to know 
+         how many variables are going to be within the array before you create it.
+         This is because arrays are immutable data structures which means after they are created their <b>length</b> can't be changed.
+    </MudText>
+
+    <Section IsSubSection="true" Title="The Rules">
+        <MudText>When dealing with an array there are four main things to keep in mind:</MudText>
+        <InfoGrid>
+            <GridSquare>Arrays have a <b>pre-defined</b> length. You must give an array a length before you are able to implement it.
+            </GridSquare>
+            <GridSquare>Arrays are <b>immutable</b>. Once the length an array is set, you cannot change it.
+            </GridSquare>
+            <GridSquare>Arrays <b>cannot</b> mix variables types.
+            </GridSquare>
+            <GridSquare>An array's index always begins at 0.
+            </GridSquare>
+        </InfoGrid>
+    </Section>
+</Section>
+
+<Section Title="Our First Array">
```

Comment:
Take a look at the "Recipes" I added to "The Basics". Let me know if you think it makes sense to have 2 new recipes: 1. Array Literal (predefined arrays) and 2. Instantiated Arrays.

Reply:
That might be a good way to arrange the arrays rather than a big grouping.

Reply:
Maybe rename this section `Creating an Array`